### PR TITLE
[Docs] Add npu build.md and update roi_pool_npu adpater

### PR DESCRIPTION
**Pull request by momo609**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: docs/zh_cn/get_started/build.md [hunk 1]
```diff
@@ -354,3 +354,51 @@ y = torch.tensor([1, 5, 3]).mlu()
...
 output = sigmoid_focal_loss(x, y, 2.0, 0.25, w, 'none')
 ```
+
+### 在昇腾 NPU 机器编译 mmcv
+
+在编译 mmcv 前，需要安装 torch_npu，完整安装教程详见 [PyTorch 安装指南](https://gitee.com/ascend/pytorch/blob/master/docs/zh/PyTorch%E5%AE%89%E8%A3%85%E6%8C%87%E5%8D%97/PyTorch%E5%AE%89%E8%A3%85%E6%8C%87%E5%8D%97.md#pytorch%E5%AE%89%E8%A3%85%E6%8C%87%E5%8D%97)
+
+#### 选项 1: 使用 NPU 设备源码编译安装 mmcv (推荐方式)
+
+- 拉取 [MMCV 源码](https://github.com/open-mmlab/mmcv/tree/master)
```
Comment (zhouzaida) [inline docs/zh_cn/get_started/build.md:364]:
```suggestion
- 拉取 [MMCV 源码](https://github.com/open-mmlab/mmcv/tree/main)
```

File: docs/zh_cn/get_started/build.md [hunk 2]
```diff
@@ -354,3 +354,51 @@ y = torch.tensor([1, 5, 3]).mlu()
...
+
+```bash
+git pull https://github.com/open-mmlab/mmcv/tree/master
```
Comment (zhouzaida) [inline docs/zh_cn/get_started/build.md:367]:
```suggestion
git pull https://github.com/open-mmlab/mmcv/tree/main
```

File: docs/zh_cn/get_started/build.md [hunk 3]
```diff
@@ -354,3 +354,51 @@ y = torch.tensor([1, 5, 3]).mlu()
...
+```
+
+#### 选项 2: 使用 pip 安装 Ascend 编译版本的 mmcv
+
+Ascend 编译版本的 mmcv-lite 在 mmcv >= 1.7.0 时已经支持直接 pip 安装
+
+```bash
+pip install mmcv-full -f https://download.openmmlab.com/mmcv/dist/ascend/torch1.8.0/index.html
```
Comment (zhouzaida) [inline docs/zh_cn/get_started/build.md:375]:
```suggestion
pip install mmcv -f https://download.openmmlab.com/mmcv/dist/ascend/torch1.8.0/index.html
```

Comment (momo609) [reply docs/zh_cn/get_started/build.md]: Modified

File: docs/zh_cn/get_started/build.md [hunk 4]
```diff
@@ -354,3 +354,51 @@ y = torch.tensor([1, 5, 3]).mlu()
...
+
+- 拉取 [MMCV 源码](https://github.com/open-mmlab/mmcv/tree/master)
+
+```bash
+git pull https://github.com/open-mmlab/mmcv/tree/master
+```
+
+#### 选项 2: 使用 pip 安装 Ascend 编译版本的 mmcv
+
+Ascend 编译版本的 mmcv-lite 在 mmcv >= 1.7.0 时已经支持直接 pip 安装
```
[the remark this answers is not in the document]
Comment (momo609) [reply docs/zh_cn/get_started/build.md]: Modified

**Pull request merged**